runtime/android/core/src/org/xwalk/core/XWalkLibraryDelegate.java

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2014 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package org.xwalk.core;
+
+import android.content.Context;
+
+import junit.framework.Assert;
+
+class XWalkLibraryDelegate {
+    protected Context mContext;
+    protected XWalkLibraryListener mLibraryListener;
+
+    public static XWalkLibraryDelegate newInstance(Context context, XWalkLibraryListener listener) {
+        if (XWalkCoreWrapper.getInstance() == null) {
+            Assert.fail("Must call XWalkCoreWrapper.init() first");
+        }
+        if (XWalkCoreWrapper.getInstance().isSharedMode()) {
+            return new XWalkLibraryDelegate(context, listener);
+        }
+        return new XWalkLibraryDelegateEmbedded(context, listener);
+    }
+
+    protected XWalkLibraryDelegate(Context context, XWalkLibraryListener listener) {
+        mContext = context;
+        mLibraryListener = listener;
+    }
+
+    public boolean initLibrary() {
+        if (mLibraryListener != null) mLibraryListener.onXWalkLibraryPrepared();
+        return true;
```

Comment:
Here is just defining the default operation, do nothing and return true. It may return false in XWalkLibraryDelegateEmbedded.